test/algorithms/classifiers/test_neural_network_classifier.py

```diff
@@ -150,9 +165,12 @@ def parity(x):
         initial_point = np.array([0.5] * ansatz.num_parameters)
 
         # construct classifier
-        classifier = NeuralNetworkClassifier(
-            qnn, optimizer=optimizer, loss=loss, initial_point=initial_point
-        )
+        if optimizer is None:
+            classifier = NeuralNetworkClassifier(qnn, loss=loss, initial_point=initial_point)
+        else:
+            classifier = NeuralNetworkClassifier(
+                qnn, optimizer=optimizer, loss=loss, initial_point=initial_point
+            )
```

Comment:
Same as above, I don't see a need for the `if` statement

Reply:
Agreed